tests/unit/Service/ApiServiceTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace OCA\Text\Tests;
+
+use OCA\Text\Service\ApiService;
+use OCA\Text\Service\ConfigService;
+use OCA\Text\Service\SessionService;
+use OCA\Text\Service\DocumentService;
+use OCA\Text\Service\EncodingService;
+use OCP\IL10N;
+use OCP\Files\File;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+class ApiServiceTest extends \PHPUnit\Framework\TestCase {
+	private ApiService $apiService;
+
+	private IRequest $request;
+	private ConfigService $configService;
+	private SessionService $sessionService;
+	private DocumentService $documentService;
+	private EncodingService $encodingService;
+	private LoggerInterface $loggerInterface;
+	private IL10N $l10n;
+	private string $userId;
+
+	public function setUp(): void {
+		$this->request = $this->createMock(IRequest::class);
+		$this->configService = $this->createMock(ConfigService::class);
+		$this->sessionService = $this->createMock(SessionService::class);
+		$this->documentService = $this->createMock(DocumentService::class);
+		$this->encodingService = $this->createMock(EncodingService::class);
+		$this->loggerInterface = $this->createMock(LoggerInterface::class);
+		$this->l10n = $this->createMock(IL10N::class);
+		$this->userId = 'admin';
+
+		$this->apiService = new ApiService(
+			$this->request,
+			$this->configService,
+			$this->sessionService,
+			$this->documentService,
+			$this->encodingService,
+			$this->loggerInterface,
+			$this->l10n,
+			$this->userId,
+			null,
+		);
+	}
+
+	public function testCreateNewSession() {
+		$file = $this->createMock(\OCP\Files\File::class);
+		$storage = $this->createMock(\OCP\Files\Storage\IStorage::class);
+		$document = $this->createMock(\OCA\Text\Db\Document::class);
+		$file->method('getStorage')->willReturn($storage);
+		$file->method('getId')->willReturn(1234);
+		$document->method('getId')->willReturn(1234);
```

Comment:
This currently fails with:
```
1) OCA\Text\Tests\ApiServiceTest::testCreateNewSession
PHPUnit\Framework\MockObject\MethodCannotBeConfiguredException: Trying to configure method "getId" which cannot be configured because it does not exist, has not been specified, is final, or is static

/var/www/html/apps/text/tests/unit/Service/ApiServiceTest.php:56

```